src/sentry/api/endpoints/project_users.py

```diff
@@ -28,12 +29,28 @@ def get(self, request, project):
             pieces = request.GET["query"].strip().split(":", 1)
             if len(pieces) != 2:
                 return Response([])
-            try:
-                queryset = queryset.filter(
-                    **{f"{EventUser.attr_from_keyword(pieces[0])}__icontains": pieces[1]}
-                )
-            except KeyError:
-                return Response([])
+            # username and ip can return multiple eventuser objects
+            if any(p in "ip" or p in "username" for p in pieces):
+                try:
+                    queryset = queryset.filter(
+                        project_id=project.id,
+                        **{f"{EventUser.attr_from_keyword(pieces[0])}__icontains": pieces[1]},
```

Comment:
We should remove the `__icontains` here, since this prevents us from using the index and making this fast.

Also just noting that by doing so we're changing from this being a search to an exact match. This endpoint isn't documented from what I can see, so I think this change it fine.